### PR TITLE
Update Product API Definitions

### PR DESCRIPTION
**Pull request by msm1992**
## Purpose
Related 4.0.0 PR : https://github.com/wso2/docs-apim/pull/3970

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 1]
```diff
@@ -6658,50 +8019,23 @@ components:
...
+            - PRIVATE
+            - RESTRICTED
           x-otherScopes:
             - apim:api_publish
-        deploymentEnvironments:
+        visibleRoles:
           type: array
-          description: |
-            List of selected deployment environments and clusters
+          description: The user roles that are able to access the API in Devportal
```
Comment (Mariangela) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:8031]:
```suggestion
          description: The user roles that are able to access the API in the Developer Portal
```

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 2]
```diff
@@ -7706,6 +9208,267 @@ components:
...
+    SubscriptionPolicy:
+      title: Subscription Throttling Policy
+      allOf:
+        - required:
+            - defaultLimit
+          type: object
+          properties:
+            policyId:
+              type: integer
+              description: Id of policy
```
Comment (Mariangela) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:9371]:
```suggestion
              description: ID of the policy
```

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 3]
```diff
@@ -7706,6 +9208,267 @@ components:
...
+              type: string
+              description: Description of the policy
+              example: Allows 30 request per minute
+            isDeployed:
+              type: boolean
+              description: Indicates whether the policy is deployed successfully or not.
+              default: false
+            tenantId:
+              type: integer
+              description: Throttling policy tenant domain id
```
Comment (Mariangela) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:9399]:
```suggestion
              description: Throttling policy tenant domain ID
```

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 4]
```diff
@@ -7706,6 +9208,267 @@ components:
...
+                $ref: '#/components/schemas/CustomAttribute'
+            stopOnQuotaReach:
+              type: boolean
+              description: |
+                This indicates the action to be taken when a user goes beyond the allocated quota. If checked, the user's requests will be dropped. If unchecked, the requests will be allowed to pass through.
+              default: false
+            billingPlan:
+              type: string
+              description: |
+                define whether this is Paid or a Free plan. Allowed values are FREE or COMMERCIAL.
```
Comment (Mariangela) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:9434]:
```suggestion
                Defines whether this is Paid or a Free plan. Allowed values are FREE or COMMERCIAL.
```

**Review by Mariangela — CHANGES_REQUESTED** [review]

**Comment by Mariangela** [comment]
@msm1992 , please apply the new review comment changes to 4.0.0 as well.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 5]
```diff
@@ -1801,272 +1803,385 @@ paths:
...
+          multipart/form-data:
+            schema:
+              properties:
+                file:
+                  type: string
+                  description: Definition to upload as a file
+                  format: binary
+                url:
+                  type: string
+                  description: Definition url
```
[the remark this answers is not in the document]
Comment (msm1992) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:2161]: should be corrected to 'API Definition URL'

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 6]
```diff
@@ -1293,504 +1393,406 @@ paths:
...
+    put:
       tags:
-        - Validation
-      summary: Check Given API Context Name already Exists
+        - API Revisions
+      summary: Update Deployment
       description: |
-        Using this operation, you can check a given API context is already used. You need to provide the context name you want to check.
-      operationId: validateAPI
+        Update deployment devportal visibility
```
[the remark this answers is not in the document]
Comment (msm1992) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:1598]: Should be 'Update deployment'

File: en/docs/reference/product-apis/devportal-apis/devportal-v2/devportal-v2.yaml [hunk 7]
```diff
@@ -1084,11 +1175,103 @@ paths:
...
+          source: curl -k "https://localhost:9443/api/am/store/v2/apis/e93fb282-b456-48fc-8981-003fb89086ae/comments/d4cf1704-5d09-491c-bc48-4d19ce6ea9b4/replies"
+      x-examples:
+        $ref: docs/examples/apis/comment_id-comments.yaml#/get
+
+  /apis/{apiId}/topics:
+    get:
+      tags:
+        - Topics
+      summary: |
+        Get a list of available topics for a given async API
```
[the remark this answers is not in the document]
Comment (msm1992) [reply en/docs/reference/product-apis/devportal-apis/devportal-v2/devportal-v2.yaml]: Updated to 'Get a list of available topics for a given Streaming API'

**Review by Mariangela — APPROVED** [review]

**Comment by Mariangela** [comment]
@msm1992 , please make sure that these changes are in 400 as well

**Pull request merged**